lib/json-schema/json-schema-loader.ts

```diff
@@ -0,0 +1,78 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import path = require("path");
+import jsonSchemaResolverPath = require("./json-schema-resolver");
+
+export class JsonSchemaLoader implements IJsonSchemaLoader {
+
+	private schemasFolderPath: string = null;
+	private schemas: IDictionary<ISchema> = null;
+	private schema: ISchema = null;
+	private loadedSchemas: IDictionary<ISchema> = null;
+
+	public constructor(private $fs: IFileSystem,
+		private $errors: IErrors,
+		private $injector: IInjector,
+		private $resources: IResourceLoader) {
+		this.schemasFolderPath = this.$resources.resolvePath("json-schemas");
+		this.schemas = Object.create(null);
+		this.loadedSchemas = Object.create(null);
+
+		this.loadSchemas().wait();
+
+		var schemaResolver = this.$injector.resolve(jsonSchemaResolverPath.JsonSchemaResolver, { schemas: this.loadedSchemas });
+		this.$injector.register("jsonSchemaResolver", schemaResolver);
+	}
+
+	private loadSchemas(): IFuture<void> {
+		return (() => {
+			var fileNames = this.$fs.readDirectory(this.schemasFolderPath).wait();
+			_.each(fileNames, (fileName: string) => {
+				if( path.extname(fileName) === ".json") {
+					var fullFilePath = path.join(this.schemasFolderPath, fileName);
+					var schema = this.$fs.readJson(fullFilePath).wait();
+					this.schemas[schema.id] = schema;
+				}
+			});
+
+			var schemas = _.values(this.schemas);
+			_.each(schemas, (schema: ISchema) => this.loadSchema(schema).wait());
```

Comment:
Can we make these calls async? For example
```Typescript
var futs: IFuture[] =[];
_.each(schemas, (schema: ISchema) => {
     var f = this.loadSchema(schema);
     futs.push(f);
});

Future.wait(futs);
```
It just looks like there are some heavy operations here.